Portal/src/Datahub.Portal/Pages/Workspace/Toolbox/WorkspaceToolboxPage.razor

```diff
@@ -168,59 +187,200 @@
+        _logger.LogInformation("Removing {Tool} from {WorkspaceAcronym}", GetLabel(tool), WorkspaceAcronym);
+        _toolCurrentDevelopmentStatusMap[tool] = ProjectResourceStatus.SendingDeletion;
+        await InvokeAsync(StateHasChanged);
+        try
+        {
+
+            var currentUser = await _userInformationService.GetCurrentPortalUserAsync();
+
+            await using var ctx = await _contextFactory.CreateDbContextAsync();
+            var _resourceCheck = await ctx.Project_Resources2
+                .FirstOrDefaultAsync(r => r.ResourceType.EndsWith(tool));
+
+
+            // adding deletion request to queue
+            var userEmail = await _userInformationService.GetUserEmail();
+            var workspaceDefinition = await _resourceMessagingService.GetWorkspaceDefinition(WorkspaceAcronym, requestingUserEmail: userEmail);
+            //await _resourceMessagingService.SendToTerraformDeleteResourceQueue(workspaceDefinition, _workspace.Project_ID, tool);
+
+            _logger.LogInformation("Tool {Tool} successfully removed from workspace {WorkspaceAcronym}", tool, WorkspaceAcronym);
+            _toolCurrentDevelopmentStatusMap[tool] = ProjectResourceStatus.Available;
+            _snackbar.Add(Localizer["{0} successfully removed from your workspace", GetLabel(tool)], Severity.Success);
+
+            _resourceCheck.Status = ProjectResourceStatus.Available.ToString();
+
+            await HandleSendEmail(tool);
+
+            await ctx.SaveChangesAsync();
+
+            await InvokeAsync(StateHasChanged);
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(e, "Error removing tool {Tool} from workspace {WorkspaceAcronym}", tool, WorkspaceAcronym);
+            _toolCurrentDevelopmentStatusMap[tool] = ProjectResourceStatus.Error;
+            _snackbar.Add(Localizer["Error removing {0} from your workspace", GetLabel(tool)], Severity.Error);
+        }
+    }
+
+    private async void GoToResourceDeletionPage(string tool)
+    {
+        await using var ctx = await _contextFactory.CreateDbContextAsync();
+        var _resourceCheck = await ctx.Project_Resources2
+            .FirstOrDefaultAsync(r => r.ResourceType.EndsWith(tool));
+        _resourceCheck.Status = ProjectResourceStatus.PendingDeletion.ToString();
+
+        await ctx.SaveChangesAsync();
+        await InvokeAsync(StateHasChanged);
+
+        NavigationManager.NavigateTo(_resourceDeletionUrl);
+    }
+
+
+
+    private async Task HandleSendEmail(string tool)
+    {
+        List<string> recipients = GetProjectAdminsEmails();
+        string subject = tool + " deleted from workspace " + WorkspaceAcronym;
+
+        string body = "An admin in your workspace (" + WorkspaceAcronym + ") has deleted the " + tool + " resource.";
+
+        await SendEmailAsync(async () =>
+        {
+            await _emailService.SendToRecipients("ssc.datahub.donotreply-nepasrepondre.datahub.spc@ssc-spc.gc.ca", recipients, subject, body);
+        });
```

Comment:
At the top of the page, you should inject:

```csharp
@inject DatahubPortalConfiguration _datahubPortalConfiguration
```

Then here, in the email service, you should use the local configurations instead of manually writing the email:

```csharp
await _emailService.SendToRecipients(_datahubPortalConfiguration.EmailNotification.SenderAddress, recipients, subject, body);
```

This way if we ever change the email we won't have to go around the whole codebase